lib/HLSL/HLOperationLower.cpp

```diff
@@ -4162,18 +4179,19 @@ Value *TranslateBufLoad(ResLoadHelper &helper, HLResource::Kind RK,
   if (isBool || (is64 && isTyped))
     EltTy = Builder.getInt32Ty();
 
-  // 64-bit types are stored as int32 pairs in typed buffers.
+  // Adjust number of components as needed.
   if (is64 && isTyped) {
+    // 64-bit types are stored as int32 pairs in typed buffers.
     DXASSERT(NumComponents <= 2, "Typed buffers only allow 4 dwords.");
     NumComponents *= 2;
+  } else if (opcode == OP::OpCode::RawBufferVectorLoad) {
+    // Native vector loads only have a single vector element in ResRet.
+    EltTy = VectorType::get(EltTy, NumComponents);
```

Comment:
This has the effect of subtly changing the alignment of vector load/store operations.  I do not think this is intended.

`GetBufLoadArgs` uses `min(alignment, LdSize)` for the alignment, where `LdSize` is now initialized to the size of the vector, instead of the size of the vector element.

Reply:
Here are a couple example vectors where the alignment would change:
- `float16_t2` or `float16_t3` in `ByteAddressBuffer` will align by 4 instead of 2.
- `float16_t4` in `StructuredBuffer` will align by 8 instead of 2.
- `float2` in `StructuredBuffer` will align by 8 instead of 4.

Test appears to have evidence of this change when using the new vector op.

Reply:
BTW: I think the store may be ok in this regard.  This only appears to impact the load.